### PR TITLE
feat: adds select and related styles

### PR DESCRIPTION
**Pull request by eljefe223**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

image

#### Description of changes

Adds `fluent-select` `fluent-listbox` and `fluent-option`

#### Focus areas to test

(optional)

### CODE REVIEW

**Review by chrisdholt — COMMENTED** [review]
Thanks @eljefe223! It looks like a couple things are missing here:

1) We need to export these from the root of the package
2) We need to run the API report

Can you make those changes?

**Review by radium-v — APPROVED** [review]

**Comment by scomea** [comment]
Did a smoke.  Saw some issues in high-contrast with no visible focus while tabbing through the trigger elements and options grayed out in the menu:
image

**Review by chrisdholt — APPROVED** [review]

**Comment by eljefe223** [comment]
> Did a smoke. Saw some issues in high-contrast with no visible focus while tabbing through the trigger elements and options grayed out in the menu:
> image

Got it, no way for me to test HC, maybe @khamudom can make a suggestion here?

File: packages/web-components/src/listbox-option/listbox-option.styles.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+`.withBehaviors(
+  accentFillActiveBehavior,
+  accentFillHoverBehavior,
+  accentFillSelectedBehavior,
+  accentForegroundCutRestBehavior,
+  forcedColorsStylesheetBehavior(
+    css`
+      :host {
```
Comment (khamudom) [inline packages/web-components/src/listbox-option/listbox-option.styles.ts:112]: also add to the :host 
`color: ${SystemColors.ButtonText};`

File: packages/web-components/src/select/select.styles.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            :host([disabled]:hover) {
+                background: ${SystemColors.ButtonFace};
+            }
+
+            :host([disabled]) .control {
+                color: ${SystemColors.GrayText};
+                border-color: ${SystemColors.GrayText};
+            }
+
+            :host(:${focusVisible}) ::slotted([aria-selected="true"][role="option"]) {
```
Comment (khamudom) [inline packages/web-components/src/select/select.styles.ts:207]: this selector should be,
`:host(:${focusVisible}) ::slotted([aria-selected="true"][role="option"]:not([disabled]))`

File: packages/web-components/src/select/select.styles.ts [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        flex: 0 0 auto;
+    }
+
+`.withBehaviors(
+  accentFillHoverBehavior,
+  accentForegroundCutRestBehavior,
+  neutralOutlineActiveBehavior,
+  neutralOutlineHoverBehavior,
+  neutralOutlineRestBehavior,
+  forcedColorsStylesheetBehavior(
```
Comment (khamudom) [inline packages/web-components/src/select/select.styles.ts:188]: Also add these rulesets. These will fix hover and focus on the select component and the opened listbox.
```css
            :host(:not([disabled]):hover) {
              background: ${SystemColors.ButtonFace};
              border-color: ${SystemColors.Highlight};
            }

            :host(:${focusVisible}) {
              forced-color-adjust: none;
              background: ${SystemColors.ButtonFace};
              border-color: ${SystemColors.Highlight};
              box-shadow: 0 0 0 1px inset ${SystemColors.Highlight};
              color: ${SystemColors.ButtonText};
              fill: currentcolor;
            }

            :host([open]) .listbox {
                background: ${SystemColors.ButtonFace};
                border: 1px solid ${SystemColors.ButtonText};
            }
```

**Comment by scomea** [comment]
smoked again, looks good.

**Review by scomea — APPROVED** [review]

**Review by khamudom — APPROVED** [review]

**Pull request merged**